src/Agents.jl

```diff
@@ -4,6 +4,8 @@ using Distributed
 using LightGraphs
 using DataFrames
 using Random
+using OpenStreetMapX
+using Parameters
```

Comment:
Julia provides the same macro as Parameters: `Base.@kwdef` and thus Parameters are not necessary as a dependency.

Reply:
I use `@with_kw` because of pretty-printing which makes life nicer when doing agent-based computing.
Consider  `Base.@kwdef struct B; x::Int; y::Vector{Int}; end` vs `@with_kw struct C; x::Int; y::Vector{Int}; end`:
```
julia> B(1, [2,3,4])
B(1, [2, 3, 4])

julia> C(1, [2,3,4])
C
  x: Int64 1
  y: Array{Int64}((3,)) [2, 3, 4]
```
I always use this for complex structures because it allows debugging in the REPL. However this is up to you. The constructors in my code can live without it altogether.